nova_vm/src/heap/heap_bits.rs

```diff
@@ -1063,3 +1066,30 @@ pub(crate) fn sweep_heap_elements_vector_descriptors<T>(
         descriptors.insert(new_key, descriptor);
     }
 }
+
+#[cfg(feature = "array-buffer")]
+pub(crate) fn sweep_data_view_side_table_values<V: Copy>(
+    // TODO: Figure out a generic way to handle all types which wrap `BaseIndex`
+    side_table: &mut AHashMap<DataView, V>,
+    compaction_list: &CompactionList,
+) {
+    let mut moved_byte_lengths = side_table
+        .iter()
+        .map(|(&data_view, &value)| {
+            let data_view_index = data_view.0.into_u32_index();
+            (
+                data_view_index,
+                data_view_index - compaction_list.get_shift_for_index(data_view_index),
+                value,
+            )
+        })
+        .filter(|(old_index, new_index, _)| old_index != new_index)
+        .collect::<Vec<_>>();
+
+    moved_byte_lengths.sort_by_key(|(old_index, _, _)| *old_index);
+
+    for (old_index, new_index, value) in moved_byte_lengths {
+        side_table.insert(DataViewIndex::from_u32_index(new_index).into(), value);
```

Comment:
**note:** Because we're not handling removals we end up with extra items remaining in the hashmap and I think we can also get dead ones incorrectly overriding existing ones, corrupting the buffer size of offset.